compiler/rustc_smir/src/stable_mir/ty.rs

```diff
@@ -242,3 +242,35 @@ pub struct BoundTy {
     pub var: usize,
     pub kind: BoundTyKind,
 }
+
+pub type Bytes = Vec<u8>;
+pub type Size = usize;
+pub type Prov = usize;
+pub type Align = u64;
+pub type InitMaskMaterialized = Vec<usize>;
+
+#[derive(Clone, Debug)]
+pub struct ProvenanceMap {
+    pub ptrs: Vec<(Size, Prov)>,
+}
```

Comment:
This will not be very useful without doc comments explaining what this is all about.

Reply:
I added comments from rustc, I don't know specifics of what smir user would want to know from this. I think it should suffice for now? If not I would love to learn more and add more information/documentation.